drivers/overlay/overlay.go

```diff
@@ -475,19 +591,37 @@ func (d *Driver) Get(id string, mountLabel string) (s string, err error) {
 	defer func() {
 		if err != nil {
 			if c := d.ctr.Decrement(mergedDir); c <= 0 {
-				syscall.Unmount(mergedDir, 0)
+				unix.Unmount(mergedDir, 0)
 			}
 		}
 	}()
 
 	workDir := path.Join(dir, "work")
 	opts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", newlowers, path.Join(id, "diff"), path.Join(id, "work"))
```

Comment:
Since we're trying to avoid the fork/exec when possible, and that means there's no child process to change its working directory to a place where relative paths resolve correctly, the last two parameters here need to be absolute paths (`diffDir` and `workDir`, respectively).  Likewise, at line 578, `lower` needs to become an absolute path, so it needs to be set to `newpath` instead of `l`.

Reply:
Fixed